libs/remix-ui/workspace/src/lib/actions/index.ts

```diff
@@ -44,19 +59,38 @@ export const initWorkspace = (filePanelPlugin) => async (reducerDispatch: React.
       dispatch(setCurrentWorkspace('code-sample'))
       const filePath = await loadWorkspacePreset('code-template')
       plugin.on('editor', 'editorMounted', async () => await plugin.fileManager.openFile(filePath))
-    } else {
-      if (workspaces.length === 0) {
-        await createWorkspaceTemplate('default_workspace', 'remixDefault')
-        plugin.setWorkspace({ name: 'default_workspace', isLocalhost: false })
-        dispatch(setCurrentWorkspace('default_workspace'))
-        await loadWorkspacePreset('remixDefault')
-      } else {
-        if (workspaces.length > 0) {
-          workspaceProvider.setWorkspace(workspaces[workspaces.length - 1])
-          plugin.setWorkspace({ name: workspaces[workspaces.length - 1], isLocalhost: false })
-          dispatch(setCurrentWorkspace(workspaces[workspaces.length - 1]))
+    } else if (window.location.pathname && window.location.pathname !== '/') {
+      const route = window.location.pathname
+      if (route.startsWith('/address/0x') && route.length === 51) {
+        const contractAddress = route.split('/')[2]
+        try {
+          const etherscanKey = await plugin.call('config', 'getAppParameter', 'etherscan-access-token')
+          if (!etherscanKey) throw new Error('unable to try fetching the source code from etherscan: etherscan access token not found. please go to the Remix settings page and provide an access token.')
+          const networks = [
+            {id: 1, name: 'mainnet'},
+            {id: 3, name: 'ropsten'},
+            {id: 4, name: 'rinkeby'},
+            {id: 42, name: 'kovan'},
+            {id: 5, name: 'goerli'}
+          ]
+          plugin.call('notification', 'toast', `Looking for contract address ${contractAddress} on different networks`)
+          for (const network of networks) {
+            const target = `/${network.name}`
+            const data = await fetchContractFromEtherscan(plugin, network, contractAddress, target)
```

Comment:
if `data` doesn't return anything you can `continue`

Reply:
done